src/Mapping.php

```diff
@@ -618,4 +618,60 @@ private function dispatch(string $event, array $data, $args = [])
             call_user_func_array($hook, $args);
         }
     }
+
+    /**
+     * Checks if string is a valid SQL column name and has no table prefix.
+     *
+     * @param string $column
+     * @return bool
+     */
+    private function requiresPrefix(string $column): bool
+    {
+        $pattern = '/^[a-zA-Z_][a-zA-Z0-9_]*$/';
+        return (bool) preg_match($pattern, $column);
+    }
+
+    /**
+     * Appends table name to provided value. Works with strings, arrays, and dictionaries. If multidimensional array is
+     * passed, the values will be updated recursively. If a dictionary is passed, only the
+     * first level of dictionary keys will be modified, does not work recursively with nested dictionaries. It is also
+     * protected from multiple passes and will only prefix the column name once.
+     *
+     * String Example:
+     *      'field' -> 'table.field'
+     *
+     * Array Example:
+     *      ['field', 'field2'] -> ['table.field', 'table.field2']
+     *      ['field', ['field2', 'field3']] -> ['table.field', ['table.field2', 'table.field3']]
+     *
+     * Dictionary Example:
+     *      ['field' => 'value'] -> ['table.field' => 'value']
+     *      ['field' => ['field2' => 'value']] -> ['table.field' => ['field2' => 'value']]
+     *      [['field' => 'value']] -> [['table.field' => 'value']]
+     *      [['field' => ['field2' => 'value']]] -> [['table.field' => ['field2' => 'value']]]
+     *
+     * @return string | array
+     */
+    private function prefixTableNameTo($input) {
+        $table = $this->definition->getTable();
+
+        if (is_string($input)) {
+            return $this->requiresPrefix($input) ? "$table.$input" : $input;
+        } elseif (is_array($input)) {
+            $output = [];
+
+            foreach ($input as $key => $value) {
+                if (is_string($key)) {
+                    $key = $this->prefixTableNameTo($key);
```

Comment:
I couldn't find anywhere that this function is applied to a dictionary. If we're not applying it to dictionary keys, we should remove this code to make it more readable.